### PR TITLE
add new datasource and some optimization

### PR DESCRIPTION
**Pull request by uncleGen**
- add jdbc datasource
- add hbase datasource
- optimize loghub datasource impl

### CODE REVIEW

File: emr-jdbc/src/main/scala/org/apache/spark/sql/aliyun/jdbc2/JdbcSink.scala [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+        //Truncate the executor table
+        truncateTable(conn, executorOptions)
+      })
+    }
+  }
+
+
+  def saveMode(outputMode: OutputMode): SaveMode = {
+    if (outputMode == OutputMode.Append()) {
```
Comment (wenxuanguan) [inline emr-jdbc/src/main/scala/org/apache/spark/sql/aliyun/jdbc2/JdbcSink.scala]: unused function

File: emr-logservice/src/main/java/org/apache/spark/streaming/aliyun/logservice/LoghubClientAgent.java [hunk 2]
```diff
@@ -213,6 +213,9 @@ public GetHistogramsResponse GetHistograms(String project, String logStore, int
           retry += 1;
           currentException = e;
         } else {
+          if (e.GetErrorCode().equals("IndexConfigNotExist")) {
+            LOG.warn("Please enable index service for " + project + "/" + logStore, e);
```
Comment (wenxuanguan) [inline emr-logservice/src/main/java/org/apache/spark/streaming/aliyun/logservice/LoghubClientAgent.java]: duplicate exception log

File: emr-logservice/src/main/scala/org/apache/spark/sql/aliyun/logservice/LoghubOffsetReader.scala [hunk 3]
```diff
@@ -189,18 +222,36 @@ object LoghubOffsetReader extends Logging with Serializable {
...
+      val accessKeySecret = sourceOptions.getOrElse("access.key.secret",
+        throw new MissingArgumentException("Missing access key secret (='access.key.secret')."))
+      val endpoint = sourceOptions.getOrElse("endpoint",
+        throw new MissingArgumentException("Missing log store endpoint (='endpoint')."))
       logServiceClient = new LoghubClientAgent(endpoint, accessKeyId, accessKeySecret)
     }
     logServiceClient
   }
 
+  def getOrCreateLogProducer(sourceOptions: Map[String, String]): LogProducer = {
```
Comment (wenxuanguan) [inline emr-logservice/src/main/scala/org/apache/spark/sql/aliyun/logservice/LoghubOffsetReader.scala]: seem not suitable to defined here

Comment (uncleGen) [reply emr-logservice/src/main/scala/org/apache/spark/sql/aliyun/logservice/LoghubOffsetReader.scala]: it is ok

**Pull request merged**